server/sonar-server/src/main/java/org/sonar/server/settings/ws/ValuesAction.java

```diff
@@ -0,0 +1,213 @@
+      .setDescription("List settings values.<br>" +
+        "If no value have been set for a setting, then the default value is returned.<br>" +
+        "Either '%s' or '%s' can be provided, not both.<br> " +
+        "Requires one of the following permissions: " +
+        "<ul>" +
+        "<li>'Administer System'</li>" +
+        "<li>'Administer' rights on the specified component</li>" +
+        "</ul>", PARAM_COMPONENT_ID, PARAM_COMPONENT_KEY)
+      .setResponseExample(getClass().getResource("values-example.json"))
+      .setSince("6.1")
+      .setHandler(this);
+    addComponentParameters(action);
+    action.createParam(PARAM_KEYS)
+      .setDescription("List of property keys")
+      .setRequired(true)
+      .setExampleValue("sonar.technicalDebt.hoursInDay,sonar.dbcleaner.cleanDirectory");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    writeProtobuf(doHandle(request), request, response);
+  }
+
+  private ValuesWsResponse doHandle(Request request) {
+    DbSession dbSession = dbClient.openSession(true);
+    try {
+      ComponentDto component = settingsWsComponentParameters.getComponent(dbSession, request);
+      settingsWsComponentParameters.checkAdminPermission(component);
+      Set<String> keys = new HashSet<>(request.mandatoryParamAsStrings(PARAM_KEYS));
+
+      ValuesWsResponse.Builder valuesBuilder = ValuesWsResponse.newBuilder();
+      new SettingsBuilder(dbSession, valuesBuilder, keys, component).build();
+      return valuesBuilder.build();
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private class SettingsBuilder {
+    private final DbSession dbSession;
+    private final ComponentDto component;
+
+    private final ValuesWsResponse.Builder valuesWsBuilder;
+    private final Set<String> keys;
+
+    private final Map<String, Settings.Setting.Builder> settingsBuilderByKey = new HashMap<>();
+
+    SettingsBuilder(DbSession dbSession, ValuesWsResponse.Builder valuesWsBuilder, Set<String> keys, @Nullable ComponentDto component) {
+      this.dbSession = dbSession;
+      this.valuesWsBuilder = valuesWsBuilder;
+      this.keys = keys;
+      this.component = component;
+    }
+
+    void build() {
+      processDefinitions();
+      processProperties(settingsLoader.loadGlobalSettings(dbSession, keys));
+      if (component != null) {
+        processProperties(settingsLoader.loadComponentSettings(dbSession, keys, component).values());
+      }
+      settingsBuilderByKey.values().forEach(Settings.Setting.Builder::build);
+    }
+
+    private void processDefinitions() {
+      propertyDefinitions.getAll().stream()
+        .filter(definition -> keys.contains(definition.key()))
+        .filter(defaultProperty -> !isNullOrEmpty(defaultProperty.defaultValue()))
+        .map(Setting::new)
+        .forEach(this::processDefaultValue);
+    }
+
+    private void processDefaultValue(Setting setting) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "processDefaultValue" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)